airbyte-config/models/src/main/java/io/airbyte/config/helpers/LogClientSingleton.java

```diff
@@ -136,33 +133,38 @@ public static void deleteLogs(Configs configs, String logPath) {
       throw new NotImplementedException("Local log deletes not supported.");
     }
     var logConfigs = new LogConfigDelegator(configs);
-    createCloudClientIfNull(logConfigs);
     var cloudLogPath = JOB_LOGGING_CLOUD_PREFIX + logPath;
     logClient.deleteLogs(logConfigs, cloudLogPath);
   }
 
   public static void setJobMdc(Path path) {
-    if (shouldUseLocalLogs(new EnvConfigs())) {
+    var configs = new EnvConfigs();
+    if (shouldUseLocalLogs(configs)) {
       LOGGER.debug("Setting docker job mdc");
       MDC.put(LogClientSingleton.JOB_LOG_PATH_MDC_KEY, path.resolve(LogClientSingleton.LOG_FILENAME).toString());
     } else {
       LOGGER.debug("Setting kube job mdc");
+      var logConfigs = new LogConfigDelegator(configs);
```

Comment:
by always creating the client at the beginning, we make sure this errors out on start up.